### PR TITLE
[pass] Revise FuseBatchNormWithConvPass

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit revises FuseBatchNormWithConvPass pass for static analysis tool.

Related: #13806 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp [hunk 1]
```diff
@@ -52,84 +70,37 @@ bool fused_batch_norm_with_conv(luci::CircleAdd *add)
...
 
   luci::CircleConst *filter = dynamic_cast<luci::CircleConst *>(conv->filter());
   luci::CircleConst *bias = dynamic_cast<luci::CircleConst *>(conv->bias());
-
   // If filter or bias of conv is not const, this pass cannot be applied.
   if (filter == nullptr || bias == nullptr)
     return false;
+  // TODO Support more data type
+  if (filter->dtype() != loco::DataType::FLOAT32)
+    return false;
```
Comment (jinevening) [inline compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:94]: It would be possible to use CHECK_OR_FALSE

Comment (jinevening) [reply compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:94]: The rest looks good to me!

File: compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp [hunk 2]
```diff
@@ -15,12 +15,30 @@
...
 #include <luci/Profile/CircleNodeOrigin.h>
 
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
 namespace
 {
+
+bool has_same_shape(luci::CircleNode *node, std::vector<uint32_t> shape)
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:30]: maybe
```suggestion
bool has_same_shape(const luci::CircleNode *node, const std::vector<uint32_t> &shape)
```

File: compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp [hunk 3]
```diff
@@ -15,12 +15,30 @@
...
+{
+  const auto rank = node->rank();
+  CHECK_OR_FALSE(rank == shape.size());
+  for (uint32_t idx = 0; idx < rank; idx++)
+  {
+    CHECK_OR_FALSE(node->dim(idx).value() == shape.at(idx));
+  }
+
+  return true;
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:39]: (optional, minor)
```suggestion
  const auto rank = node->rank();
  CHECK_OR_FALSE(rank == shape.size());
  for (uint32_t idx = 0; idx < rank; idx++)
    CHECK_OR_FALSE(node->dim(idx).value() == shape.at(idx));
  return true;
```

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by jinevening — APPROVED** [review]

**Pull request merged**